source/secure_tunneling.c

```diff
@@ -4,291 +4,324 @@
  */
 
 #include <aws/iotdevice/private/secure_tunneling_impl.h>
+#include <aws/iotdevice/private/secure_tunneling_operations.h>
 
+#include <aws/common/clock.h>
 #include <aws/common/string.h>
 #include <aws/http/proxy.h>
 #include <aws/http/request_response.h>
 #include <aws/http/websocket.h>
 #include <aws/io/channel_bootstrap.h>
 #include <aws/io/event_loop.h>
 #include <aws/io/socket.h>
-#include <aws/iotdevice/private/iotdevice_internals.h>
 #include <aws/iotdevice/private/serializer.h>
+#include <inttypes.h>
 #include <math.h>
 
+#ifdef _MSC_VER
+#    pragma warning(push)
+#    pragma warning(disable : 4232) /* function pointer to dll symbol */
+#endif
+
 #define MAX_WEBSOCKET_PAYLOAD 131076
 #define INVALID_STREAM_ID 0
 #define PAYLOAD_BYTE_LENGTH_PREFIX 2
+#define MIN_RECONNECT_DELAY_MS 1000
+#define MAX_RECONNECT_DELAY_MS 120000
 #define PING_TASK_INTERVAL ((uint64_t)20 * 1000000000)
+#define WEBSOCKET_HEADER_NAME_ACCESS_TOKEN "access-token"
+#define WEBSOCKET_HEADER_NAME_CLIENT_TOKEN "client-token"
+#define WEBSOCKET_HEADER_NAME_PROTOCOL "Sec-WebSocket-Protocol"
+#define WEBSOCKET_HEADER_PROTOCOL_VALUE "aws.iot.securetunneling-2.0"
```

Comment:
This will become 3.0 as part of my next task. I may end up making a lot of these strings more permanent instead of concatenating them at that point as well.